src/main/java/business/controllers/UserController.java

```diff
@@ -0,0 +1,185 @@
+        if (email == null) {
+            return new ResponseEntity<Object>("No email address entered.", HttpStatus.BAD_REQUEST);
+        }
+        if (user.getUsername() == null || !user.getUsername().equals(email)) {
+            // check for uniqueness (also enforced by database):
+            User u = userRepository.findByUsernameAndDeletedFalse(email);
+            if (u == null) {
+                user.setUsername(email);
+            } else {
+                return new ResponseEntity<Object>("Email address not available.", HttpStatus.NOT_MODIFIED);
+            }
+        }
+        return null;
+    }
+
+    private ResponseEntity<Object> createNewUser(ProfileRepresentation body) {
+        if (body.getPassword1() != null && body.getPassword1().equals(body.getPassword2()))
+        {
+            if (userRepository.findByUsername(body.getUsername()) != null ) {
+                throw new IllegalArgumentException("Credentials already exist in our system.");
+            }
+
+            Role role = roleRepository.findByName(body.getCurrentRole());
+            Set<Role> roles;
+            if (role == null) {
+                return new ResponseEntity<Object>("No role selected.", HttpStatus.BAD_REQUEST);
+            } else {
+                roles = Collections.singleton(role);
+            }
+
+            User user = new User(body.getUsername(), body.getPassword1(), true, roles);
+
+            ResponseEntity<Object> result = transferUserData(body, user);
+            if (result != null) {
+                return result;
+            }
+            return new ResponseEntity<Object>(new ProfileRepresentation(userRepository.save(user)), HttpStatus.OK);
+        }
+        else
+        {
+            return new ResponseEntity<Object>("Passwords do not match.", HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    @RequestMapping(value = "/admin/users", method = RequestMethod.POST)
+    public ResponseEntity<Object> create(Principal principal, @RequestBody ProfileRepresentation body) {
+        LogFactory.getLog(getClass()).info("POST /admin/users (for user: " + principal.getName() + ")");
+        return createNewUser(body);
+    }
+
+    @RequestMapping(value = "/admin/users/{id}", method = RequestMethod.PUT,
+            consumes = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<Object> update(Principal principal, @PathVariable Long id, @RequestBody ProfileRepresentation body) {
+        LogFactory.getLog(getClass()).info("PUT /admin/users/" + id);
+        User user = userRepository.findOne(id);
+        if (user != null) {
+            ResponseEntity<Object> result = transferUserData(body, user);
+            if (result != null) {
+                return result;
+            }
+            return new ResponseEntity<Object>(new ProfileRepresentation(userRepository.save(user)), HttpStatus.OK);
+        }
+        return new ResponseEntity<Object>("User not found.", HttpStatus.NOT_FOUND);
```

Comment:
Not to meddle too much on your project, but:
- Many methods take a principal argument. It looks like could just be injected into the resource.
- The users should probably be a sub-resource.
- You have similar logging statements in all the methods. It's almost certain there's already jersey class that logs the same thing (and if not there are better ways to accomplish the same thing, like filters -- see JerseyLogginFilter -- or using spring to wrap the beans).
- Related: you use commons logging...
- Returning `ResponseEntity` directly prevents usage of the more interesting jax-rs features, like automatically generating responses based on exception throws or arbitrary objects returned (by registering the appropriate renderers, or whatever they're called in jax-rs).